sdk/core/scheduler/common.h

```diff
@@ -155,4 +162,58 @@ namespace sched
 	void exception_entry_asm(void);
 	__END_DECLS
 
+	template<typename T>
+	struct HeapObject
+	{
+		class Deleter
+		{
+			struct SObjStruct *heapCapability;
+
+			public:
+			Deleter(struct SObjStruct *heapCapability)
+			  : heapCapability(heapCapability)
+			{
+			}
+			void operator()(T *object)
+			{
+				object->~T();
+				free(object);
```

Comment:
Should that be `heap_free(heapCapability, object)`?

Reply:
Yup, that's in the next commit.